File: 09. Graph/01. Breath First Search (BFS)/BFS.py
from collections import deque

def BFS(node, adj, vis):
    q = deque()
    q.append(node)
    vis[node] = True

    while q:
        cur = q.popleft()
        print(cur, end=" ")

        for neigh in adj[cur]:
            if not vis[neigh]:
                q.append(neigh)
                vis[neigh] = True

def main():
    n = 6
    adj = [[] for _ in range(n)]
    adj[0] = [1, 2]
    adj[1] = [0, 3, 4]
    adj[2] = [0, 4]
    adj[3] = [1]
    adj[4] = [1, 2, 5]
    adj[5] = [4]

    vis = [False] * n

    print("BFS starting from node 0: ", end="")
    BFS(0, adj, vis)

main()

File: 09. Graph/01. Breath First Search (BFS)/BFS.go
package main

import (
	"fmt"
)

func BFS(node int, adj [][]int, vis []bool) {
	queue := []int{}
	queue = append(queue, node)
	vis[node] = true

	for len(queue) > 0 {
		cur := queue[0]
		queue = queue[1:]

		fmt.Print(cur, " ")

		for _, neigh := range adj[cur] {
			if !vis[neigh] {
				queue = append(queue, neigh)
				vis[neigh] = true
			}
		}
	}
}

func main() {
	n := 6
	adj := make([][]int, n)

	adj[0] = []int{1, 2}
	adj[1] = []int{0, 3, 4}
	adj[2] = []int{0, 4}
	adj[3] = []int{1}
	adj[4] = []int{1, 2, 5}
	adj[5] = []int{4}

	vis := make([]bool, n)

	fmt.Print("BFS starting from node 0: ")
	BFS(0, adj, vis)
}